### ISSUE TITLE
Cannot start container due to iptable problem

### ISSUE THREAD

**Issue by ditisjordy** [bubuntux/nordvpn#52]
Trying to run NordVPN in bridge mode on a Synology NAS. Using recommended parameters I get the following error:

![image](https://user-images.githubusercontent.com/19697295/72928786-e06ede00-3d58-11ea-94e5-adac0b11fb4b.png)

Am I the only one?

Parameters (COUNTRY = NL makes no difference):

![image](https://user-images.githubusercontent.com/19697295/72928966-38a5e000-3d59-11ea-8a1d-6e50c873f522.png)

**Comment by bubuntux** [comment]
may have to do with Synology itself and not this image, new bin version seems to fix this type of issues with Synology please take a look at #47  for more info

**Comment by aidan-walden** [comment]
I'm having what appears to be a similar issue on Ubuntu desktop (amd64)

```
vpn            | Staring firewall...

vpn            | Adding network route 192.168.1.0/24...

vpn            | Whitelisting api.nordvpn.com...

vpn            | iptables v1.8.3 (legacy): host/network `api.nordvpn.com' not fo                                                                                                                                                                          und

vpn            | Try `iptables -h' or 'iptables --help' for more information.

vpn            | Selecting the best server...

vpn            | Searching for technology: openvpn_udp

vpn            | Unable to find a server with the specified parameters, using any recommended server                                                                                                                                                                          

vpn            | Best server :

vpn            | Using config file /vpn/ovpn/fi82.nordvpn.com.udp.ovpn...

vpn            | Connecting ...

vpn            | + sg vpn -c 'openvpn --config /vpn/ovpn/fi82.nordvpn.com.udp.ovpn --auth-user-pass /vpn/auth --auth-nocache                                 --s                                                                                                                                                                             cript-security 2 --up /etc/openvpn/up.sh --down /etc/openvpn/down.sh                                                                                                                                                                                                              --pull-filter ignore "ping-restart" --ping-exit 180'                                                                                                                                                                             

vpn            | Thu Jan 23 00:49:19 2020 OpenVPN 2.4.7 x86_64-alpine-linux-musl[SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on May  5 2019                             

vpn            | Thu Jan 23 00:49:19 2020 library versions: OpenSSL 1.1.1d  10 Sep 2019, LZO 2.10    

vpn            | Thu Jan 23 00:49:19 2020 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts                                                                                                                                                                             

vpn            | Thu Jan 23 00:49:19 2020 Outgoing Control Channel Authenticatio                                                                                                                                                                             n: Using 512 bit message hash 'SHA512' for HMAC authentication

vpn            | Thu Jan 23 00:49:19 2020 Incoming Control Channel Authentication: Using 512 bit message hash 'SHA512' for HMAC authentication   

vpn            | Thu Jan 23 00:49:19 2020 TCP/UDP: Preserving recently used remote address [AF_INET]196.196.203.43:1194

vpn            | Thu Jan 23 00:49:19 2020 Socket Buffers: R=[212992->212992] S=[212992->212992]                                                                                                                      

vpn            | Thu Jan 23 00:49:19 2020 UDP link local: (not bound)

vpn            | Thu Jan 23 00:49:19 2020 UDP link remote: [AF_INET]196.196.203.                                                                                                                                                                             43:1194
```

And when using bin:
```
Attaching to vpn
Technology is successfully set to 'NordLynx'.
Subnet 172.21.0.0/16 is whitelisted successfully.
Please enter your login details.
vpn            | Stdin/Stdout should be terminal
vpn            | 2020/01/23 01:05:09 listen unix /run/nordvpnd.sock: bind: address already in use
Whoops! Cannot reach System Daemon.
vpn            | 2020/01/23 01:05:21 listen unix /run/nordvpnd.sock: bind: address already in use
Whoops! Cannot reach System Daemon.
vpn            | 2020/01/23 01:05:32 listen unix /run/nordvpnd.sock: bind: address already in use
Whoops! Cannot reach System Daemon.
vpn            | 2020/01/23 01:05:44 listen unix /run/nordvpnd.sock: bind: address already in use
Whoops! Cannot reach System Daemon.
vpn            | 2020/01/23 01:05:55 listen unix /run/nordvpnd.sock: bind: address already in use
Whoops! Cannot reach System Daemon.
vpn            | 2020/01/23 01:06:07 listen unix /run/nordvpnd.sock: bind: address already in use
```
Parameters:
![image](https://user-images.githubusercontent.com/44129920/72960163-c8a85180-3d7a-11ea-8981-e2176d77753c.png)

**Comment by bubuntux** [comment]
can you provide the docker run command (not docker compose) and log, please include env variable `DEBUG=on` just be sure to remove user and pss.

**Comment by aidan-walden** [comment]
Command:
`docker run -ti --cap-add=NET_ADMIN --device /dev/net/tun --name nordvpn -e USER=myemail -e PASS='mypass'-e CONNECT=United_States -e TECHNOLOGY=NordLynx -e DEBUG=on -d bubuntux/nordvpn`

Log:
```

Staring firewall...
Whitelisting downloads.nordcdn.com...
Downloading config files...
Whitelisting api.nordvpn.com...
Selecting the best server...
Best server : us3311.nordvpn.com
Using config file /vpn/ovpn/us3311.nordvpn.com.udp.ovpn...
Connecting ...
+ sg vpn -c 'openvpn --config /vpn/ovpn/us3311.nordvpn.com.udp.ovpn --auth-user-pass /vpn/auth --auth-nocache                                 --script-security 2 --up /etc/openvpn/up.sh --down /etc/openvpn/down.sh                                 '
Fri Jan 24 01:24:54 2020 OpenVPN 2.4.7 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on May  5 2019
Fri Jan 24 01:24:54 2020 library versions: OpenSSL 1.1.1d  10 Sep 2019, LZO 2.10
Fri Jan 24 01:24:54 2020 WARNING: --ping should normally be used with --ping-restart or --ping-exit
Fri Jan 24 01:24:54 2020 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
Fri Jan 24 01:24:54 2020 Outgoing Control Channel Authentication: Using 512 bit message hash 'SHA512' for HMAC authentication
Fri Jan 24 01:24:54 2020 Incoming Control Channel Authentication: Using 512 bit message hash 'SHA512' for HMAC authentication
Fri Jan 24 01:24:54 2020 TCP/UDP: Preserving recently used remote address: [AF_INET]23.226.131.235:1194
Fri Jan 24 01:24:54 2020 Socket Buffers: R=[212992->212992] S=[212992->212992]
Fri Jan 24 01:24:54 2020 UDP link local: (not bound)
Fri Jan 24 01:24:54 2020 UDP link remote: [AF_INET]23.226.131.235:1194
Fri Jan 24 01:24:54 2020 TLS: Initial packet from [AF_INET]23.226.131.235:1194, sid=5752df2b 81893d9a
Fri Jan 24 01:24:54 2020 VERIFY OK: depth=2, C=PA, O=NordVPN, CN=NordVPN Root CA
Fri Jan 24 01:24:54 2020 VERIFY OK: depth=1, C=PA, O=NordVPN, CN=NordVPN CA4
Fri Jan 24 01:24:54 2020 VERIFY KU OK
Fri Jan 24 01:24:54 2020 Validating certificate extended key usage
Fri Jan 24 01:24:54 2020 ++ Certificate has EKU (str) TLS Web Server Authentication, expects TLS Web Server Authentication
Fri Jan 24 01:24:54 2020 VERIFY EKU OK
Fri Jan 24 01:24:54 2020 VERIFY OK: depth=0, CN=us3311.nordvpn.com
Fri Jan 24 01:24:54 2020 Control Channel: TLSv1.2, cipher TLSv1.2 ECDHE-RSA-AES256-GCM-SHA384, 4096 bit RSA
Fri Jan 24 01:24:54 2020 [us3311.nordvpn.com] Peer Connection Initiated with [AF_INET]23.226.131.235:1194
Fri Jan 24 01:24:55 2020 SENT CONTROL [us3311.nordvpn.com]: 'PUSH_REQUEST' (status=1)
Fri Jan 24 01:24:57 2020 PUSH: Received control message: 'PUSH_REPLY,redirect-gateway def1,dhcp-option DNS 103.86.96.100,dhcp-option DNS 103.86.99.100,sndbuf 524288,rcvbuf 524288,explicit-exit-notify,comp-lzo no,route-gateway 10.8.0.1,topology subnet,ping 60,ping-restart 180,ifconfig 10.8.0.8 255.255.255.0,peer-id 32,cipher AES-256-GCM'
Fri Jan 24 01:24:57 2020 OPTIONS IMPORT: timers and/or timeouts modified
Fri Jan 24 01:24:57 2020 OPTIONS IMPORT: explicit notify parm(s) modified
Fri Jan 24 01:24:57 2020 OPTIONS IMPORT: compression parms modified
Fri Jan 24 01:24:57 2020 OPTIONS IMPORT: --sndbuf/--rcvbuf options modified
Fri Jan 24 01:24:57 2020 Socket Buffers: R=[212992->425984] S=[212992->425984]
Fri Jan 24 01:24:57 2020 OPTIONS IMPORT: --ifconfig/up options modified
Fri Jan 24 01:24:57 2020 OPTIONS IMPORT: route options modified
Fri Jan 24 01:24:57 2020 OPTIONS IMPORT: route-related options modified
Fri Jan 24 01:24:57 2020 OPTIONS IMPORT: --ip-win32 and/or --dhcp-option options modified
Fri Jan 24 01:24:57 2020 OPTIONS IMPORT: peer-id set
Fri Jan 24 01:24:57 2020 OPTIONS IMPORT: adjusting link_mtu to 1657
Fri Jan 24 01:24:57 2020 OPTIONS IMPORT: data channel crypto options modified
Fri Jan 24 01:24:57 2020 Data Channel: using negotiated cipher 'AES-256-GCM'
Fri Jan 24 01:24:57 2020 Outgoing Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
Fri Jan 24 01:24:57 2020 Incoming Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
Fri Jan 24 01:24:57 2020 ROUTE_GATEWAY 172.17.0.1/255.255.0.0 IFACE=eth0 HWADDR=02:42:ac:11:00:02
Fri Jan 24 01:24:57 2020 TUN/TAP device tun0 opened
Fri Jan 24 01:24:57 2020 TUN/TAP TX queue length set to 100
Fri Jan 24 01:24:57 2020 /sbin/ip link set dev tun0 up mtu 1500
Fri Jan 24 01:24:57 2020 /sbin/ip addr add dev tun0 10.8.0.8/24 broadcast 10.8.0.255
Fri Jan 24 01:24:57 2020 /etc/openvpn/up.sh tun0 1500 1585 10.8.0.8 255.255.255.0 init
Fri Jan 24 01:24:57 2020 /sbin/ip route add 23.226.131.235/32 via 172.17.0.1
Fri Jan 24 01:24:57 2020 /sbin/ip route add 0.0.0.0/1 via 10.8.0.1
Fri Jan 24 01:24:57 2020 /sbin/ip route add 128.0.0.0/1 via 10.8.0.1
Fri Jan 24 01:24:57 2020 Initialization Sequence Completed

```

Log again, this time using bin:
```
+ NET_IFACE=eth0
+ [[ -n '' ]]
+ [[ '' =~ ^[0-9]+$ ]]
+ kill_switch
+ iptables -F OUTPUT
+ ip6tables -F OUTPUT
+ iptables -P OUTPUT DROP
+ ip6tables -P OUTPUT DROP
+ iptables -A OUTPUT -o lo -j ACCEPT
+ ip6tables -A OUTPUT -o lo -j ACCEPT
++ ip -o addr show dev eth0
++ awk '$3 == "inet"  {print $4}'
++ ip -o addr show dev eth0
++ awk '$3 == "inet6" {print $4; exit}'
+ local docker_network=172.17.0.2/16 docker6_network=
+ [[ -n 172.17.0.2/16 ]]
+ iptables -A OUTPUT -d 172.17.0.2/16 -j ACCEPT
+ [[ -n '' ]]
+ iptables -A OUTPUT -m owner --gid-owner vpn -j ACCEPT
+ ip6tables -A OUTPUT -m owner --gid-owner vpn -j ACCEPT
+ ip6tables -A OUTPUT -p udp -m udp --dport 53 -j ACCEPT
+ ip6tables -A OUTPUT -p udp -m udp --dport 51820 -j ACCEPT
+ ip6tables -A OUTPUT -p tcp -m tcp --dport 1194 -j ACCEPT
+ ip6tables -A OUTPUT -p udp -m udp --dport 1194 -j ACCEPT
+ ip6tables -A OUTPUT -o eth0 -d api.nordvpn.com -j ACCEPT
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n '' ]]
+ sleep 0.75
+ sg vpn -c nordvpnd
+ nordvpn login -u aidan.walden15@gmail.com -p W@ldenX3

-
\
|
/
-
\
|
/
  
[32mWelcome to NordVPN! You can now connect to VPN by using 'nordvpn connect'.
[0m+ setup_nordvpn
+ [[ -n NordLynx ]]
+ nordvpn set technology NordLynx

-
  
[32mTechnology is successfully set to 'NordLynx'.
[0m+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n '' ]]
++ ip -o addr show dev eth0
++ awk '$3 == "inet"  {print $4}'
+ local docker_network=172.17.0.2/16
+ [[ -n 172.17.0.2/16 ]]
+ nordvpn whitelist add subnet 172.17.0.2/16

-
  
[32mSubnet 172.17.0.0/16 is whitelisted successfully.
[0m+ [[ -n on ]]
+ nordvpn settings

-
  
Technology: NordLynx
Kill Switch: disabled
CyberSec: disabled
Notify: disabled
Auto-connect: disabled
DNS: disabled
Whitelisted subnets:
	172.17.0.0/16
+ nordvpn connect United_States

-
\
|
/
-
  
[32mConnecting to United States #3395 (us3395.nordvpn.com)
[0m
\
|
/
-
\
|
/INFO: (nordvpn-us3395) 2020/01/24 01:28:13 Attaching to interface
INFO: (nordvpn-us3395) 2020/01/24 01:28:13 Interface set up
INFO: (nordvpn-us3395) 2020/01/24 01:28:13 Device started
2020/01/24 01:28:13 address :10.5.0.2/16
2020/01/24 01:28:13 [nordlynx] ip -4 address add 10.5.0.2/16 dev nordvpn-us3395
2020/01/24 01:28:13 [nordlynx] ip link set mtu 1420 up dev nordvpn-us3395
2020/01/24 01:28:13 sysctl: setting key "net.ipv6.conf.all.disable_ipv6": Read-only file system

  
[33mWe could not disable IPv6. For more information, please check activity log.
[0m2020/01/24 01:28:13 [nordlynx] ip rule add to 172.98.93.217 lookup main
2020/01/24 01:28:13 [nordlynx] ip -4 route add 0.0.0.0/0 dev nordvpn-us3395 table 51820
2020/01/24 01:28:13 [nordlynx] ip -4 rule add not fwmark 51820 table 51820
2020/01/24 01:28:13 [nordlynx] ip -4 rule add table main suppress_prefixlength 0

  
[32mYou are connected to United States #3395 (us3395.nordvpn.com)!
[0m
  
++ pidof nordvpnd
+ tail -f --pid=52 /var/log/nordvpn/daemon.log
2020/01/24 01:28:13 address :10.5.0.2/16
2020/01/24 01:28:13 [nordlynx] ip -4 address add 10.5.0.2/16 dev nordvpn-us3395
2020/01/24 01:28:13 [nordlynx] ip link set mtu 1420 up dev nordvpn-us3395
2020/01/24 01:28:13 sysctl: setting key "net.ipv6.conf.all.disable_ipv6": Read-only file system
2020/01/24 01:28:13 [nordlynx] ip rule add to 172.98.93.217 lookup main
2020/01/24 01:28:13 [nordlynx] ip -4 route add 0.0.0.0/0 dev nordvpn-us3395 table 51820
2020/01/24 01:28:13 [nordlynx] ip -4 rule add not fwmark 51820 table 51820
2020/01/24 01:28:13 [nordlynx] ip -4 rule add table main suppress_prefixlength 0

```
The bin log appears to be successful in connecting to the NordVPN server, however that container still has no internet access:

```
aidan@DellUbuntu:~/docker$ docker exec -it nordvpn bash
root@f42b4552461b:/# curl google.com
curl: (6) Could not resolve host: google.com
root@f42b4552461b:/# 
```

**Comment by aidan-walden** [comment]
Command:
`docker run -ti --cap-add=NET_ADMIN --device /dev/net/tun --name nordvpn -e USER=myemail -e PASS='mypass' -e CONNECT=United_States -e TECHNOLOGY=NordLynx -e DEBUG=on -d bubuntux/nordvpn`

Log:
```
Staring firewall...
Whitelisting downloads.nordcdn.com...
Downloading config files...
Whitelisting api.nordvpn.com...
Selecting the best server...
Best server : us3311.nordvpn.com
Using config file /vpn/ovpn/us3311.nordvpn.com.tcp.ovpn...
Connecting ... 
+ sg vpn -c 'openvpn --config /vpn/ovpn/us3311.nordvpn.com.tcp.ovpn --auth-user-pass /vpn/auth --auth-nocache                                 --script-security 2 --up /etc/openvpn/up.sh --down /etc/openvpn/down.sh                                 '
Fri Jan 24 01:36:23 2020 OpenVPN 2.4.7 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on May  5 2019
Fri Jan 24 01:36:23 2020 library versions: OpenSSL 1.1.1d  10 Sep 2019, LZO 2.10
Fri Jan 24 01:36:23 2020 WARNING: --ping should normally be used with --ping-restart or --ping-exit
Fri Jan 24 01:36:23 2020 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
Fri Jan 24 01:36:23 2020 NOTE: --fast-io is disabled since we are not using UDP
Fri Jan 24 01:36:23 2020 Outgoing Control Channel Authentication: Using 512 bit message hash 'SHA512' for HMAC authentication
Fri Jan 24 01:36:23 2020 Incoming Control Channel Authentication: Using 512 bit message hash 'SHA512' for HMAC authentication
Fri Jan 24 01:36:23 2020 TCP/UDP: Preserving recently used remote address: [AF_INET]23.226.131.235:443
Fri Jan 24 01:36:23 2020 Socket Buffers: R=[131072->131072] S=[16384->16384]
Fri Jan 24 01:36:23 2020 Attempting to establish TCP connection with [AF_INET]23.226.131.235:443 [nonblock]
Fri Jan 24 01:36:24 2020 TCP connection established with [AF_INET]23.226.131.235:443
Fri Jan 24 01:36:24 2020 TCP_CLIENT link local: (not bound)
Fri Jan 24 01:36:24 2020 TCP_CLIENT link remote: [AF_INET]23.226.131.235:443
Fri Jan 24 01:36:24 2020 TLS: Initial packet from [AF_INET]23.226.131.235:443, sid=097a3f55 e8b7ac67
Fri Jan 24 01:36:24 2020 VERIFY OK: depth=2, C=PA, O=NordVPN, CN=NordVPN Root CA
Fri Jan 24 01:36:24 2020 VERIFY OK: depth=1, C=PA, O=NordVPN, CN=NordVPN CA4
Fri Jan 24 01:36:24 2020 VERIFY KU OK
Fri Jan 24 01:36:24 2020 Validating certificate extended key usage
Fri Jan 24 01:36:24 2020 ++ Certificate has EKU (str) TLS Web Server Authentication, expects TLS Web Server Authentication
Fri Jan 24 01:36:24 2020 VERIFY EKU OK
Fri Jan 24 01:36:24 2020 VERIFY OK: depth=0, CN=us3311.nordvpn.com
Fri Jan 24 01:36:25 2020 Control Channel: TLSv1.2, cipher TLSv1.2 ECDHE-RSA-AES256-GCM-SHA384, 4096 bit RSA
Fri Jan 24 01:36:25 2020 [us3311.nordvpn.com] Peer Connection Initiated with [AF_INET]23.226.131.235:443
Fri Jan 24 01:36:26 2020 SENT CONTROL [us3311.nordvpn.com]: 'PUSH_REQUEST' (status=1)
Fri Jan 24 01:36:26 2020 PUSH: Received control message: 'PUSH_REPLY,redirect-gateway def1,dhcp-option DNS 103.86.96.100,dhcp-option DNS 103.86.99.100,sndbuf 524288,rcvbuf 524288,explicit-exit-notify,comp-lzo no,route-gateway 10.7.3.1,topology subnet,ping 60,ping-restart 180,ifconfig 10.7.3.2 255.255.255.0,peer-id 0,cipher AES-256-GCM'
Fri Jan 24 01:36:26 2020 OPTIONS IMPORT: timers and/or timeouts modified
Fri Jan 24 01:36:26 2020 OPTIONS IMPORT: --explicit-exit-notify can only be used with --proto udp
Fri Jan 24 01:36:26 2020 OPTIONS IMPORT: compression parms modified
Fri Jan 24 01:36:26 2020 OPTIONS IMPORT: --sndbuf/--rcvbuf options modified
Fri Jan 24 01:36:26 2020 Socket Buffers: R=[131072->425984] S=[87040->425984]
Fri Jan 24 01:36:26 2020 OPTIONS IMPORT: --ifconfig/up options modified
Fri Jan 24 01:36:26 2020 OPTIONS IMPORT: route options modified
Fri Jan 24 01:36:26 2020 OPTIONS IMPORT: route-related options modified
Fri Jan 24 01:36:26 2020 OPTIONS IMPORT: --ip-win32 and/or --dhcp-option options modified
Fri Jan 24 01:36:26 2020 OPTIONS IMPORT: peer-id set
Fri Jan 24 01:36:26 2020 OPTIONS IMPORT: adjusting link_mtu to 1659
Fri Jan 24 01:36:26 2020 OPTIONS IMPORT: data channel crypto options modified
Fri Jan 24 01:36:26 2020 Data Channel: using negotiated cipher 'AES-256-GCM'
Fri Jan 24 01:36:26 2020 Outgoing Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
Fri Jan 24 01:36:26 2020 Incoming Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
Fri Jan 24 01:36:26 2020 ROUTE_GATEWAY 172.17.0.1/255.255.0.0 IFACE=eth0 HWADDR=02:42:ac:11:00:02
Fri Jan 24 01:36:26 2020 TUN/TAP device tun0 opened
Fri Jan 24 01:36:26 2020 TUN/TAP TX queue length set to 100
Fri Jan 24 01:36:26 2020 /sbin/ip link set dev tun0 up mtu 1500
Fri Jan 24 01:36:26 2020 /sbin/ip addr add dev tun0 10.7.3.2/24 broadcast 10.7.3.255
Fri Jan 24 01:36:26 2020 /etc/openvpn/up.sh tun0 1500 1587 10.7.3.2 255.255.255.0 init
Fri Jan 24 01:36:26 2020 /sbin/ip route add 23.226.131.235/32 via 172.17.0.1
Fri Jan 24 01:36:26 2020 /sbin/ip route add 0.0.0.0/1 via 10.7.3.1
Fri Jan 24 01:36:26 2020 /sbin/ip route add 128.0.0.0/1 via 10.7.3.1
Fri Jan 24 01:36:26 2020 Initialization Sequence Completed
```

Log again, this time using bin:
```
+ NET_IFACE=eth0
+ [[ -n '' ]]
+ [[ '' =~ ^[0-9]+$ ]]
+ kill_switch
+ iptables -F OUTPUT
+ ip6tables -F OUTPUT
+ iptables -P OUTPUT DROP
+ ip6tables -P OUTPUT DROP
+ iptables -A OUTPUT -o lo -j ACCEPT
+ ip6tables -A OUTPUT -o lo -j ACCEPT
++ ip -o addr show dev eth0
++ awk '$3 == "inet"  {print $4}'
++ ip -o addr show dev eth0
++ awk '$3 == "inet6" {print $4; exit}'
+ local docker_network=172.17.0.2/16 docker6_network=
+ [[ -n 172.17.0.2/16 ]]
+ iptables -A OUTPUT -d 172.17.0.2/16 -j ACCEPT
+ [[ -n '' ]]
+ iptables -A OUTPUT -m owner --gid-owner vpn -j ACCEPT
+ ip6tables -A OUTPUT -m owner --gid-owner vpn -j ACCEPT
+ ip6tables -A OUTPUT -p udp -m udp --dport 53 -j ACCEPT
+ ip6tables -A OUTPUT -p udp -m udp --dport 51820 -j ACCEPT
+ ip6tables -A OUTPUT -p tcp -m tcp --dport 1194 -j ACCEPT
+ ip6tables -A OUTPUT -p udp -m udp --dport 1194 -j ACCEPT
+ ip6tables -A OUTPUT -o eth0 -d api.nordvpn.com -j ACCEPT
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n '' ]]
+ sleep 0.75
+ sg vpn -c nordvpnd
+ nordvpn login -u myemail -p mypass

-
\
|
/
-
\
|
/
  
[32mWelcome to NordVPN! You can now connect to VPN by using 'nordvpn connect'.
[0m+ setup_nordvpn
+ [[ -n NordLynx ]]
+ nordvpn set technology NordLynx

-
  
[32mTechnology is successfully set to 'NordLynx'.
[0m+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n '' ]]
+ [[ -n '' ]]
++ ip -o addr show dev eth0
++ awk '$3 == "inet"  {print $4}'
+ local docker_network=172.17.0.2/16
+ [[ -n 172.17.0.2/16 ]]
+ nordvpn whitelist add subnet 172.17.0.2/16

-
  
[32mSubnet 172.17.0.0/16 is whitelisted successfully.
[0m+ [[ -n on ]]
+ nordvpn settings

-
  
Technology: NordLynx
Kill Switch: disabled
CyberSec: disabled
Notify: disabled
Auto-connect: disabled
DNS: disabled
Whitelisted subnets:
	172.17.0.0/16
+ nordvpn connect United_States

-
\
|
/
-
  
[32mConnecting to United States #3395 (us3395.nordvpn.com)
[0m
\
|
/
-
\
|
/INFO: (nordvpn-us3395) 2020/01/24 01:28:13 Attaching to interface
INFO: (nordvpn-us3395) 2020/01/24 01:28:13 Interface set up
INFO: (nordvpn-us3395) 2020/01/24 01:28:13 Device started
2020/01/24 01:28:13 address :10.5.0.2/16
2020/01/24 01:28:13 [nordlynx] ip -4 address add 10.5.0.2/16 dev nordvpn-us3395
2020/01/24 01:28:13 [nordlynx] ip link set mtu 1420 up dev nordvpn-us3395
2020/01/24 01:28:13 sysctl: setting key "net.ipv6.conf.all.disable_ipv6": Read-only file system

  
[33mWe could not disable IPv6. For more information, please check activity log.
[0m2020/01/24 01:28:13 [nordlynx] ip rule add to 172.98.93.217 lookup main
2020/01/24 01:28:13 [nordlynx] ip -4 route add 0.0.0.0/0 dev nordvpn-us3395 table 51820
2020/01/24 01:28:13 [nordlynx] ip -4 rule add not fwmark 51820 table 51820
2020/01/24 01:28:13 [nordlynx] ip -4 rule add table main suppress_prefixlength 0

  
[32mYou are connected to United States #3395 (us3395.nordvpn.com)!
[0m
  
++ pidof nordvpnd
+ tail -f --pid=52 /var/log/nordvpn/daemon.log
2020/01/24 01:28:13 address :10.5.0.2/16
2020/01/24 01:28:13 [nordlynx] ip -4 address add 10.5.0.2/16 dev nordvpn-us3395
2020/01/24 01:28:13 [nordlynx] ip link set mtu 1420 up dev nordvpn-us3395
2020/01/24 01:28:13 sysctl: setting key "net.ipv6.conf.all.disable_ipv6": Read-only file system
2020/01/24 01:28:13 [nordlynx] ip rule add to 172.98.93.217 lookup main
2020/01/24 01:28:13 [nordlynx] ip -4 route add 0.0.0.0/0 dev nordvpn-us3395 table 51820
2020/01/24 01:28:13 [nordlynx] ip -4 rule add not fwmark 51820 table 51820
2020/01/24 01:28:13 [nordlynx] ip -4 rule add table main suppress_prefixlength 0

```

**Comment by hchapman1089** [comment]
I am having the same issues with using the new bin as well. Here is my compose 

  nord-vpn:
    # This needs to be running for deluge
    container_name: nord-vpn
    image: bubuntux/nordvpn:latest
    cap_add:
      - net_admin
    devices:
      - /dev/net/tun
    environment:
      - USER='nonono'
      - PASS='nonono'
      - COUNRTY=United States
      - TECHNOLOGY=NordLynx
      - CATEGORY=P2P
      - NETWORK=10.1.1.0/24
      - OPENVPN_OPTS=--pull-filter ignore "ping-restart" --ping-exit 180
      - TZ=America/New_York
    ports:
      - 8112:8112
    restart: unless-stopped

**Comment by bubuntux** [comment]
@hchapman1089 please follow the issue template to provide information, the compose is not enough

**Comment by hchapman1089** [comment]
@bubuntux 
When running the latest version I get an iptables error with could not load match 'owner'. Container then seems to then reboot
If I run the openvpn version I am able to connect again without any issues

**Logs**
Try `iptables -h' or 'iptables --help' for more information.
Technology is successfully set to 'NordLynx'.
Subnet 172.18.0.0/16 is whitelisted successfully.
Subnet 10.1.1.0/24 is whitelisted successfully.
Please enter your login details.
Stdin/Stdout should be terminal
iptables v1.8.2 (legacy): Couldn't load match `owner':No such file or directory

**Additional context**
running on a synology nas 
 - Linux MediaServer 4.4.59+ #24922 SMP PREEMPT Mon Aug 19 12:13:37 CST 2019 x86_64 GNU/Linux synology_apollolake_718+

$ cat /proc/net/ip_tables_matches
ipvs
mark
conntrack
conntrack
conntrack
addrtype
limit
state
addrtype
recent
recent
iprange
udplite
udp
tcp
multiport
icmp

**Comment by snieking** [comment]
Happens to me as well.

**Comment by bubuntux** [comment]
I made this image to fulfill my needs, i'm gonna keep it public and you can use it at your own risk, if something doesn't work for you, sry but is up to you to fix it, unfortunately i don't have time to keep addressing ppls issues, from now on i'm only accepting pull requests.

**Comment by bubuntux** [comment]
fixed on e0fa84966fbddc6053d72ccd4bb20d23a947234b
